kubernetes-model/src/test/java/io/fabric8/kubernetes/api/model/DeploymentTest.java

```diff
@@ -0,0 +1,42 @@
+package io.fabric8.kubernetes.api.model;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Scanner;
+
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import org.junit.Test;
+
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_ARRAY_ORDER;
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_EXTRA_FIELDS;
+import static net.javacrumbs.jsonunit.core.Option.TREATING_NULL_AS_ABSENT;
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+
+public class DeploymentTest {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    public void should_produce_same_json_from_unmarshalled_one() throws Exception {
+        // given
+        final String originalJson = loadJson("/valid-deployment.json");
+
+        // when
+        final Deployment deployment = mapper.readValue(originalJson, Deployment.class);
+        final String serializedJson = mapper.writeValueAsString(deployment);
+
+        // then
+        assertThatJson(serializedJson).when(IGNORING_ARRAY_ORDER, TREATING_NULL_AS_ABSENT, IGNORING_EXTRA_FIELDS)
+                .isEqualTo(originalJson);
+    }
+
+    private String loadJson(String path) {
```

Comment:
If this method is common across multiple test cases could we extract it into helper class?